Client/RecentItemsHelper.swift

```diff
@@ -38,6 +38,25 @@ extension BookmarkItemData: RecentlySavedItem {
     }
 }
 
+// This is an intermediary object to allow us to more easily use this data in a thread safe way.
+// Thread safety is difficult to ensure when passing classes around by reference.
+struct RecentlySavedBookmark: RecentlySavedItem {
```

Comment:
Curious to know, that struct is used for bookmarks but we keep using `ReadingListItem` for reading list. Why was that intermediary object needed for bookmarks and not for reading list?

Reply:
ReadingListItem is already a struct. It's possible the intermediary struct for the bookmarks is overkill because I suspect the biggest issue is the count of items getting out of sync and not some detail of the item itself, but this is the safest option. With a struct passing by value it means we are only ever dealing with copies of the data and there is pretty much zero chance of another thread messing with it while we are actively using it.